### ISSUE TITLE
Seems not saving the gradients successfully?

### ISSUE THREAD

**Issue by JiabangChen** [jacobgil/pytorch-grad-cam#562]
I am confused about the line 27-29, definition "save_gradient(self, module, input, output):", Class "ActivationsAndGradients" in the file "activations_and_gradients.py", as the screen shot below

![Image](https://github.com/user-attachments/assets/6e316692-0fea-44aa-b8e0-5e40e5b14fd5)

Here put a forward hook to save the gradients, but I think the "requires_grads" attribute of the output should be False because it is not a leaf tensor. So it will just return and not save the gradients. However, when I am reviewing the examples of using Grad-CAM to explain the ViT, I don't find any code that make the input image "requires_grads = True", so I am thinking it seems not saving the gradients....... Help me guys, thanks!

**Comment by JiabangChen** [comment]
SORRY....... I think I have made a terrible mistake...... So here is the new understanding, I think the output of the intermediate layer is not a leaf_node, but the "requires_grads" = True since it needs to calculate the gradient of the weights and bias (they are leaf node and requires_grads = True) of the previous layer. So, if a parameter is involved in the gradient calculation of the weights / bias / other leaf node with  "requires_grads = True", then, of course it has a positive value of "requires_grads"

**Comment by JiabangChen** [comment]
If my new understands still have some errors, pls help me to fix them : )
